### ISSUE TITLE
column pinning have some styling bugs

### ISSUE THREAD

**Issue by mburger81** [swimlane/ngx-datatable#527]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here
```

**Current behavior**
Showing your example for column pinning here http://swimlane.github.io/ngx-datatable/# in my opinion there is a style problem. The divider between **City** and **State** Header is not horizontal aligned with the divider in the rows.

![image](https://cloud.githubusercontent.com/assets/3778892/23016252/2c6bcd14-f436-11e6-9fc1-212427e9d453.png)

**Expected behavior**
I think the divider should be aligned.

**Reproduction of the problem**
See the example in our demo page

**Comment by Dedac** [comment]
Any progress on this, you can still see the error in the live pinning demo

**Comment by jackrg94** [comment]
There also seems to be an issue with pinned columns overriding the theme of the datatable. For example, I fetch my columns and rows from server and then I say, if this is a pinned column, create as pinned column in html, and for the rest of the columns, create in for loop. The pinned columns show up with white font and white background, where the rest of my columns are all themed correctly with black text and alternating row backgrounds.

**Comment by mburger81** [comment]
Some news about this bug???

**Comment by amcdnl** [comment]
This is because the scrollbar is shown. I want to remove native scrollbars and use a solution like Facebook did here - https://facebook.github.io/fixed-data-table/

**Comment by maxxt** [comment]
Any news on this one?

**Comment by mertergun** [comment]
When there isn't the vertical scroll bar, you can see the below content, as shown in [this image](https://i.imgur.com/cn0FzRV.png)

**Comment by xmeng1** [comment]
Yes, even there is no scroll bar, it is also not aligned. See #1078.  
![selection_055](https://user-images.githubusercontent.com/5753142/32060862-efe5cb62-ba67-11e7-852c-ba41be7ed46c.png)

**Comment by xmeng1** [comment]
I found a ~~best way~~ or stupid way to solve the issue temporarily, when the view is changed, find all the element with the class: `datatable-row-right datatable-row-group` and add margin-left: 15px attributes.
```
public ngAfterViewChecked() {
    const elements: HTMLElement[] =
        this.elRef.nativeElement.getElementsByClassName('datatable-row-right datatable-row-group');
    if (elements && elements.length > 0) {
      for (const obj of elements) {
        obj.style.setProperty('margin-left', '15px');
      }
    }
  }
```

**Comment by jkon** [comment]
Similar issue I am seeing is that when using bootstrap or dark theme, pinned columns don't hide scrolled columns beneath like material theme does. You can see the issue by changing the theme on the pinning demo (http://swimlane.github.io/ngx-datatable/#pinning) using the dev console. If you change it to bootstrap, the middle columns scroll horizontally and you can still see them under the frozen columns
![screen shot 2017-11-22 at 3 09 00 pm](https://user-images.githubusercontent.com/392042/33151878-2a26512c-cf97-11e7-9d9d-d0fd31406e2b.png)

**Comment by cmcclellen** [comment]
Any updates on this?  Any workarounds to allow striping of the rows, yet let the unpinned columns slide under the pinned ones like material?

**Comment by harsha-nagaraj** [comment]
The above issue can be fixed by giving margin-left: -9px and use :host /deep/ to overwrite css on dataTable for the class
![screenshot from 2018-03-20 12-47-21](https://user-images.githubusercontent.com/32260075/37640916-a519cfe6-2c3d-11e8-878c-c9f566859b4b.png)

:host /deep/ .ngx-datatable.material .datatable-header .datatable-row-right{
    margin-left: -8px;
}

**Comment by tonny008** [comment]
I created a pr for this bug #1524

**Comment by HearnMichael** [comment]
Any progress on this issue?

**Comment by SanaeHanaoui** [comment]
i have the same issue yet not solved 
i am using the version  ^16.0.2

**Comment by DSplawski92** [comment]
> Similar issue I am seeing is that when using bootstrap or dark theme, pinned columns don't hide scrolled columns beneath like material theme does. You can see the issue by changing the theme on the pinning demo (http://swimlane.github.io/ngx-datatable/#pinning) using the dev console. If you change it to bootstrap, the middle columns scroll horizontally and you can still see them under the frozen columns
> ![screen shot 2017-11-22 at 3 09 00 pm](https://user-images.githubusercontent.com/392042/33151878-2a26512c-cf97-11e7-9d9d-d0fd31406e2b.png)

It seems issue still occurs. Is there any solution based on CSS?
